File: Array-String/HashTable/CPP/check_if_all_the_integers_in_a_range_are_covered.cpp
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
    public:
        bool isCovered(std::vector<std::vector<int>>& ranges, int left, int right) {
            // Use a simple array instead of std::map since keys are continuous and small
            std::vector<bool> covered(right - left + 1, false);

            for (const auto& r : ranges) {
                int start = std::max(left, r[0]);
                int end = std::min(right, r[1]);
                for (int i = start; i <= end; ++i) {
                    covered[i - left] = true;  // shift index to 0-based
                }
            }

            for (bool c : covered) {
                if (!c) return false;
            }
            return true;
        }
    };

File: Array-String/HashTable/JS/check_if_all_the_integers_in_a_range_are_covered.js
/**
 * @param {number[][]} ranges
 * @param {number} left
 * @param {number} right
 * @return {boolean}
 */
var isCovered = function(ranges, left, right) {
    const covered = new Array(right - left + 1).fill(false);
    for (const [start, end] of ranges) {
        for (let i = Math.max(start, left); i <= Math.min(end, right); i++) {
            covered[i-left] = true;
        }
    }
    return covered.every(val => val === true);
};